terraform/gcp/main.tf

```diff
@@ -0,0 +1,103 @@
+data "google_compute_zones" "available" {}
+
+resource "google_compute_instance" "wiresteward" {
+  count                     = local.instance_count
+  name                      = "${local.name}-${count.index}"
+  machine_type              = "e2-micro"
+  can_ip_forward            = true
+  zone                      = data.google_compute_zones.available.names[count.index]
+  allow_stopping_for_update = true
+
+  tags = [local.name]
+
+  boot_disk {
+    initialize_params {
+      image = "flatcar-beta"
+    }
+  }
+
+  network_interface {
+    subnetwork = var.subnet_link
+
+    access_config {
+      // Ephemeral IP
+    }
+  }
+
+  metadata = {
+    user-data = var.ignition[count.index]
+  }
+}
+
+resource "google_compute_instance_group" "wiresteward" {
+  count = local.instance_count
+  name  = "${local.name}-${count.index}"
+
+  instances = [
+    google_compute_instance.wiresteward.*.id[count.index],
+  ]
+
+  named_port {
+    name = "oauth2-http"
+    port = "4180"
+  }
+
+  zone = data.google_compute_zones.available.names[count.index]
+}
+
+resource "google_dns_record_set" "wiresteward" {
+  count = local.instance_count
+  name  = "${local.wireguard_endpoint[count.index]}."
+  type  = "A"
+  ttl   = 30 # TODO increase the value once happy with setup
```

Comment:
is this TODO left intentionally?

Reply:
Yes, once we're happy with the module we should set it to something more reasonable but the module is more than likely to change shape in the next couple weeks.